### PR TITLE
Update report-generator.ts

### PR DESCRIPTION
**Pull request by dabonnie**
Option to provide a name for generated reports

### CODE REVIEW

File: src/report-generator.ts [hunk 1]
```diff
@@ -173,8 +175,14 @@ async function run() {
     markdownReport += "\n";
     console.log(markdownReport);
     const htmlFileContent = convertMarkdownToHtml(markdownReport);
-    fs.writeFileSync('report.md', markdownReport);
-    fs.writeFileSync('report.html', htmlFileContent);
+    const fileName = core.getInput("file-name", {required: false});
```
Comment (zmichaels11) [inline src/report-generator.ts]: I'm really rusty on typescript and javascript, but I think you can do:
```javascript
const fileName = core.getInput("file-name", {required: false}} || DEFAULT_REPORT_FILE_NAME;
```

Comment (piraka9011) [reply src/report-generator.ts]: `fileName: string = ...`

File: src/report-generator.ts [hunk 2]
```diff
@@ -9,6 +9,8 @@ const GITHUB_MARKDOWN_CSS =
 
 const QUERY_STRING = "is:pr archived:false draft:false";
 
+const DEFAULT_REPORT_FILE_NAME = "report";
```
Comment (piraka9011) [inline src/report-generator.ts]: `const DEFAULT_REPORT_FILE_NAME: string = "report";`

**Comment by dabonnie** [comment]
Still failing after a local ```npm run build```, I'll have to check with @thomas-moulard if I'm missing anything.

**Comment by piraka9011** [comment]
@dabonnie checks pass :)

**Comment by dabonnie** [comment]
> Why do we need this option?

All reports use the same format and have the same name. After downloaded to my system or uploaded elsewhere, how can they be differentiated? There's an issue open to change the format for the monthly: https://github.com/ros-security/github-contribution-report-generator/issues/10.

**Comment by thomas-moulard** [comment]
I don't think that's going to give you what you want, I don't have at the top of my mind a way to expend action parameters to dates. How would you see this being used?

**Comment by dabonnie** [comment]
> I don't think that's going to give you what you want, I don't have at the top of my mind a way to expend action parameters to dates. How would you see this being used?

This PR just provides the ability for an action to specify the artifact name, or at least that's the intent.

**Comment by prajakta-gokhale** [comment]
@dabonnie do we want to move ahead with this PR?

**Comment by dabonnie** [comment]
> @dabonnie do we want to move ahead with this PR?

I'd like to sync with @thomas-moulard first in order to discuss the intent.

**Comment by piraka9011** [comment]
Closing in favor of #12

**Pull request closed without merging**